src/js/profile/indicator.js

```diff
@@ -9,62 +10,96 @@ const indicatorTitleClass = '.profile-indicator__title h4';
+        $(indicatorTitleClass, this.indicatorNode).text(title);
+        $(chartDescClass, this.indicatorNode).text(indicatorData.description);
+        this.extractMetadata(indicatorData.metadata, this.indicatorNode);
+    }
 
-        if (isLink) {
+    extractMetadata(metadata, indicatorNode) {
+        if (metadata.url) {
             let ele = $('<a></a>');
-            $(ele).text(indicatorData.metadata.source);
-            $(ele).attr('href', indicatorData.metadata.url);
+            $(ele).text(metadata.source);
+            $(ele).attr('href', metadata.url);
             $(ele).attr('target', '_blank');
             $(sourceClass, indicatorNode).html(ele);
         } else {
-            $(sourceClass, indicatorNode).text(indicatorData.metadata.source);
+            $(sourceClass, indicatorNode).text(metadata.source);
         }
+    }
+}
 
-        if (indicatorData.groups !== null && typeof indicatorData.groups !== 'undefined') {
-            for (const [group, items] of Object.entries(indicatorData.groups)) {
+export class IndicatorBlock extends InfoBlock {
+    constructor(parent, formattingConfig, wrapper, title, indicatorData, detail, _isLast) {
+        super(parent, formattingConfig, wrapper, title, indicatorData, detail, _isLast);
+        this.wrapper = wrapper;
+        this.title = title;
+        this.detail = detail;
+        this.addIndicator();
+    }
+    addIndicator() {
+
+        if (this.indicatorData.groups !== null && typeof this.indicatorData.groups !== 'undefined') {
+            for (const [group, items] of Object.entries(this.indicatorData.groups)) {
                 this.groups.push(group);
             }
         }
 
 
-        const configuration = detail.indicators[title].chartConfiguration;
-        const indicators = detail.indicators;
+        const configuration = this.detail.indicators[this.title].chartConfiguration;
+        const indicators = this.detail.indicators;
 
         let hasValues = this.indicatorData.data.some(function(e) { return e.count > 0 });
 
         if (hasValues) {
-            let c = new Chart(this, configuration, indicatorData, this.groups, indicatorNode, title);
+            let c = new Chart(this.Indicator, configuration, this.indicatorData, this.groups, this.indicatorNode, this.title);
             this.bubbleEvents(c, [
                 'profile.chart.saveAsPng', 'profile.chart.valueTypeChanged',
                 'profile.chart.download_csv', 'profile.chart.download_excel', 'profile.chart.download_json', 'profile.chart.download_kml',
                 'point_tray.subindicator_filter.filter'
             ]);
 
             if (!isLast) {
-                $(indicatorNode).removeClass('last');
+                $(this.indicatorNode).removeClass('last');
             }
 
-            wrapper.append(indicatorNode);
+            this.wrapper.append(this.indicatorNode);
+        }
+    }
+}
+
+export class MarkdownBlock extends InfoBlock {
```

Comment:
The frontend doesn't know that it's markdown - as far as the frontend knows, it's HTML